### ISSUE TITLE
Transitive reference to type not detected

### ISSUE THREAD

**Issue by skogsbaer** [etylizer/etylizer#179]
Module `m1` uses `m2` via `bar`:

```erlang
-module(m1).

-compile(export_all).
-compile(nowarn_export_all).

-spec foo() -> integer().
foo() -> m2:bar(1).
```

Module `m2` defines the spec of `bar` via `m3`:

```erlang
-module(m2).

-compile(export_all).
-compile(nowarn_export_all).

-spec bar(m3:t()) -> integer().
bar(X) -> X.
```

Here is `m3`:

```erlang
-module(m3).

-compile(export_all).
-compile(nowarn_export_all).

-export_type([t/0]).

-type t() :: integer().
```

Type checking `m1` now fails with

```erlang
./m2.erl:6:11: Name error: type m3:t/0 undefined
```

It seems as if the types from `m3` were not added to the symtab for type checking `m1`.

**Comment by skogsbaer** [comment]
This probably also causes the recompilation logic to be wrong: if `m3` changes, then `m1` must be re-checked as well.

**Comment by skogsbaer** [comment]
Fixed with #188
